procrastinate/aiopg_connector.py

```diff
@@ -97,7 +97,7 @@ def __init__(self, *, socket_timeout: float = store.SOCKET_TIMEOUT, **kwargs: An
         self.socket_timeout = socket_timeout
 
     async def get_connection(self):
-        if not self._connection:
+        if not self._connection or self._connection.closed:
```

Comment:
On the one hand, I’d rather that function wouldn’t mix logic on our side (re-opening if closed) and intimate knowledge about psycopg2 (detecting if a connection is open or closed by checking the `closed` parameter). Would you mind splitting those 2 (a `is_openned` function that takes an aiopg connection) ? I

Reply:
`self._connection` is an aiopg object!

Reply:
and `closed` is a a documented property: https://aiopg.readthedocs.io/en/stable/core.html#aiopg.Connection.closed.

Reply:
Yes, it's not about it being documented, it's about putting all the code that includes knowledge about a 3rd party in the same place.

Hm, maybe it was the "intimate" part that made my comment harder to understand :/

For me it's all about isolating 3rd party dependencies. I want to try and decouple as much as possible procrastinate from psycopg2/aiopg.

Reply:
It doesn't have to be this this way if you feel otherwise. I'm totally OK going your way. If you have specific reasons in mind to do so, it's even better.

Reply:
I must admit that I don't understand what why you'd like :)

Are you suggesting to add a `connection_is_open` function? Like this:

```py
def connection_is_open(self):
    return not self._connection.closed
```

And then? What about `get_connection`? Would you change it to create a new connection when `self.connection_is_open()` returns `False`? If not, how can one get a new connection when the current one has been closed? Should `close_connection` set `self._connection` to `None`?

Reply:
The module is split between the functions (at the top) that do interact with psycopg2 and aiopg, and the class implementation at the bottom that calls the functions, but doesn't really know about psycopg/aiopg. I'm suggesting:

```python
def connection_is_open(connection: aiopg.Connection) -> bool:
    return not connection.closed
```
and then get_connection would ask for `connection_is_open(self._connection)` instead of `self._connection.closed`.

Reply:
Ohh, got you. Sorry for being slow on this!